pkg/alerts/alertsHandler/cronJobHandler.go

```diff
@@ -75,54 +90,111 @@ func RemoveCronJob(alertId string) error {
 	return nil
 }
 
-func evaluate(alertToEvaluate *alertutils.AlertDetails, job gocron.Job) {
-	serResVal := pipesearch.ProcessAlertsPipeSearchRequest(alertToEvaluate.QueryParams)
-	if serResVal == -1 {
-		log.Errorf("ALERTSERVICE: evaluate: Empty response returned by server.")
+func updateAlertStateAndCreateAlertHistory(alertDetails *alertutils.AlertDetails, alertState alertutils.AlertState, eventDesc string) error {
+	err := updateAlertState(alertDetails.AlertId, alertState)
+	if err != nil {
+		log.Errorf("ALERTSERVICE: updateAlertStateAndCreateAlertHistory: could not update the state to %v. Alert=%+v & err=%+v.", alertState, alertDetails.AlertName, err)
+		return err
+	}
+
+	alertEvent := alertutils.AlertHistoryDetails{
+		AlertId:          alertDetails.AlertId,
+		AlertType:        alertDetails.AlertType,
+		EventDescription: eventDesc,
+		UserName:         alertutils.SystemGeneratedAlert,
+		EventTriggeredAt: time.Now().UTC(),
+	}
+	_, err = databaseObj.CreateAlertHistory(&alertEvent)
+	if err != nil {
+		log.Errorf("ALERTSERVICE: updateAlertStateAndCreateAlertHistory: could not create alert event in alert history. found error = %v", err)
+		return err
+	}
+	return nil
+}
+
+func evaluateLogAlert(alertToEvaluate *alertutils.AlertDetails, job gocron.Job) {
+	serResVal, isResultsEmpty, err := pipesearch.ProcessAlertsPipeSearchRequest(alertToEvaluate.QueryParams)
+	if err != nil {
+		log.Errorf("ALERTSERVICE: evaluateLogAlert: Error processing logs query. Alert=%+v, err=%+v", alertToEvaluate.AlertName, err)
 		return
 	}
-	isFiring := evaluateConditions(serResVal, &alertToEvaluate.Condition, alertToEvaluate.Value)
-	if isFiring {
-		err := updateAlertState(alertToEvaluate.AlertId, alertutils.Firing)
 
-		if err != nil {
-			log.Errorf("ALERTSERVICE: evaluate: could not update the state to FIRING. Alert=%+v & err=%+v.", alertToEvaluate.AlertName, err)
+	if isResultsEmpty {
+		// Should not return here, as this can mean, there are no valid logs that satisfies the Alert Query.
+		// This should be considered as a normal state. And we should update the alert state to Inactive.
+		log.Warnf("ALERTSERVICE: evaluateLogAlert: Empty response returned by server.")
```

Comment:
So why is this `log.Warnf` here?

Reply:
Previously, we were logging and returning this as an error. So I thought may be we still need to log this as a warning at least, so that users might know that their query returned an empty response, this is not necessarily an error on the system side, but an error on the user side, where they might have entered a wrong query.